### DISCUSSION TITLE
Add a way to source machine specfic config files

### DISCUSSION THREAD

**Discussion by Kn4ughty** [hyprwm/Hyprland#8470] [General]
In sway, you use commands inside of your include paths, like this:
`include $HOME/.config/sway/$(hostname)/*.conf`.
This means you can import different configs depending on which device you are using.
Some way of doing this in hyprland would be very nice.

# Use case
This is useful in the case where you sync your files between computers, but want some differences. Say for example on your laptop you want to disable some of the more expensive graphical effects. Or you want a different wallpaper for each machine.

**Comment by fufexan** [comment]
I was thinking a `hyprland.conf.d/` dir with auto-loaded files would be more useful. Then you would only have to place the needed files for each config, and it would be handled automatically.

**Reply by izmyname** [reply]
That would be cool as, if I understand correctly, it would be used as full replacement to hyprland.conf, without having to manually source files in it. Having just `hyprland.conf.d/` with drop-in files, like `00-monitors.conf`, `10-environments.conf` etc.

**Reply by Kn4ughty** [reply]
If you just want to include a bunch of files dynamically you can do that already by putting something like this in your config somewhere. 
`source = ~/.config/hypr/hyprland.conf.d/*.conf`

**Reply by izmyname** [reply]
Oh, thanks for the info. I sourced each one, inividually, 

Although, having hyprland sourcing them ootb, without hyprland.conf file, would still be a cool feature

**Reply by Soulsuke** [reply]
That works, but it's the kind of workaround which requires a certain amount of user intervantion when managing dotfiles...  

A way to use env variables or a hyprctl command to source files later on would help quite a lot.

**Reply by fufexan** [reply]
@Soulsuke are you referring to my comment or @Kn4ughty's?

**Reply by Soulsuke** [reply]
Be it a default folder to drop files into or wildcard which sources everything from a custom source path, that requires manual files management. It's a great way not to have a single huge config file instead of a bunch of smaller ones, but something like @Kn4ughty originaly suggested would allow to only source certain file depending on the hostname, with no further intervention

**Comment by tjoneseng** [comment]
This was #1416, fixed in b03c897 but broken in 13f6f0b.
